File: domain/user/src/main/java/whenyourcar/domain/user/serviceImpl/UserCommonServiceImpl.java
package whenyourcar.domain.user.serviceImpl;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.servlet.http.HttpSession;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import whenyourcar.domain.common.code.exception.AuthenticationException;
import whenyourcar.domain.common.code.status.ErrorStatus;
import whenyourcar.domain.user.dto.security.SessionUser;
import whenyourcar.domain.user.service.UserCommonService;
import whenyourcar.storage.mysql.data.entity.User;
import whenyourcar.storage.mysql.repository.user.UserCommonRepository;

@Service
@RequiredArgsConstructor
public class UserCommonServiceImpl implements UserCommonService {
    private final UserCommonRepository userCommonRepository;
    @Override
    public SessionUser getSessionUser(String email)  {
        User user =  userCommonRepository.findByEmail(email)
                .orElseThrow(() -> new AuthenticationException(ErrorStatus.USER_IS_NOT_EXIST));
        return SessionUser.builder()
                .userId(user.getId())
                .email(user.getEmail())
                .name(user.getName())
                .picture(user.getPicture())
                .build();
    }

    @Override
    public User getUser(HttpSession httpSession) {
        SessionUser user = (SessionUser) httpSession.getAttribute("user");
        return userCommonRepository.findUserById(user.getUserId())
                .orElseThrow(() -> new AuthenticationException(ErrorStatus.USER_IS_NOT_EXIST));
    }
}

File: application/src/main/kotlin/howmuchcar/application/service/user/UserCommonServiceImpl.kt
package howmuchcar.application.service.user

import howmuchcar.application.port.`in`.user.UserCommonUseCase
import howmuchcar.application.port.out.db.user.UserPort
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class UserCommonServiceImpl(
    private val userPort: UserPort
) : UserCommonUseCase {

    override fun getUserById(userId: Long): User {
        val user = userPort.findUserById(userId)
        return user
    }
}